lsvmload/log.c

```diff
@@ -171,35 +179,52 @@ EFI_STATUS PutLog(
         va_end(ap);
     }
 
+    /* Get the time stamp to the log and convert to string. */
+    {
+        time_t t = time(NULL) - first;
+
+        /* Format is "[NUM]: ". Work our way backwards. */
+        *--numStart = '\0';
+        *--numStart = ' ';
+        *--numStart = ':';
+        *--numStart = ']';
+        do {
+            *--numStart = '0' + (t % 10);
+            t /= 10;
+        }
+        while (t > 0);
+        *--numStart = '[';
+    }
+
     /* Convert the string to single-character (ignore special characters) */
     {
+        /* Log format is "LOGLEVEL: [TIMESTAMP]: MSG" */
+        UINTN levelLen = Strlen(logLevelStr);
+        UINTN numLen = Strlen(numStart);
+        UINTN msgLen = StrLen(wcs);
         UINTN i;
-        UINTN len = StrLen(wcs);
 
-        if (!(str = Malloc(sizeof(char) * (len + 2))))
+        /* +2 for the ": " after LOGLEVEL and +2 for the ending "\n\0" */
+        if (!(str = Malloc(sizeof(char) * (levelLen + numLen + msgLen + 4))))
         {
             status = EFI_OUT_OF_RESOURCES;
             goto done;
         }
 
-        for (i = 0; i < len; i++)
```

Comment:
It looks like this could all be replaced with the ``VPrint`` library function.

Reply:
Thanks! I'll fix it up with VSPrint(). Should be a lot clearer.